attach/mutt_attach.c

```diff
@@ -487,6 +489,26 @@ int mutt_view_attachment(FILE *fp, struct Body *a, enum ViewAttachMode mode,
 
     mutt_rfc3676_space_unstuff_attachment(a, mutt_buffer_string(tmpfile));
 
+    /* check for multipart/related and save attachments with a Content-ID */
+    struct Body *related_ancestor = NULL;
+    bool has_related = false;
+    struct CidMapList cid_map_list;
+    STAILQ_INIT(&cid_map_list);
```

Comment:
We can reduce the scope of `cid_map_list`, too.
And optimise the initialisation:
```suggestion
    struct CidMapList cid_map_list = STAILQ_HEAD_INITIALIZER(&cid_map_list);
```